models/demos/yolov4/web_demo/server/fast_api_yolov4.py

```diff
@@ -0,0 +1,70 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+import json
+from fastapi import FastAPI, File, UploadFile
+from io import BytesIO
+from PIL import Image
+from models.demos.yolov4.tests.yolov4_perfomant_webdemo import Yolov4Trace2CQ
+import ttnn
+
+import cv2
+import numpy as np
+import torch
+import time
+
+app = FastAPI(
+    title="YOLOv4 object detection",
+    description="Inference engine to detect objects in image.",
+    version="0.0",
+)
+
+
+@app.get("/")
+async def root():
+    return {"message": "Hello World"}
+
+
+@app.on_event("startup")
+async def startup():
+    device_id = 0
+    device = ttnn.CreateDevice(device_id, l1_small_size=24576, trace_region_size=1617920, num_command_queues=2)
+    ttnn.enable_program_cache(device)
+    global model
+    model = Yolov4Trace2CQ()
+    model.initialize_yolov4_trace_2cqs_inference(device)
```

Comment:
Question, but do you see better perf with this trace 2cqs case compared to just trace? It looks like since we're blocking/synchronizing every execution, then we won't see any benefit of multi-cq since we can't really overlap inputs/outputs and might be additional overhead over just using trace

Reply:
Thanks @tt-aho for the feedback.
We haven't tried the trace-only flow, let me try it.
So, do you recommend trace-only or trace-multi-cq without synchronization?
Will the second option be viable to our application of client/server flow?

Reply:
It's possible to use trace-multi-cq for an application like this, but it's slightly more involved since you'd want to be able to queue multiple inputs before waiting for an output if possible, and have host wait for a specific output/event instead of blocking on an entire cq to complete. This is somewhat described here in this comment https://github.com/tenstorrent/tt-metal/issues/14475#issuecomment-2452854324 but we haven't formalized it with examples/descriptions in the tech report yet.

For the current implementation where we are blocking every iteration I would expect single cq to perform the same/slightly better.

Reply:
Yes I think that's right, since the bottleneck wouldn't be our model in this case.